### PR TITLE
Create Main Code

### PR DESCRIPTION
**Pull request by sounisi5011**
- [x] Add tests for basic features
- [x] Add tests for edge case
- [x] Add source codes
- [x] Write document

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1 +1,92 @@
...
+            descList.map(([, desc]) => desc.value).filter(isNotPrimitive),
+        );
+
+        this.__objectMap.set(value, { descMap, childObjectSet });
+        for (const childObject of childObjectSet) {
+            this.__set(childObject);
+        }
+    }
+
+    private __rollback(
```
[the remark this answers is not in the document]
Comment (sounisi5011) [reply src/index.ts]: Resolved by https://github.com/sounisi5011/object-rollback/pull/1/commits/15c90eca56accfedb5ee6c1ba5d7946301ff7f2d
